src/Data/Functor/Listable.hs

```diff
@@ -0,0 +1,56 @@
+module Data.Functor.Listable
+( Listable(..)
+, cons0
+, cons1
+, cons2
+, cons3
+, cons4
+, cons5
+, cons6
+, (\/)
+, Listable1(..)
+, tiers1
+, Listable2(..)
+, tiers2
+, liftCons1
+, liftCons2
+, liftCons3
+) where
+
+import Test.LeanCheck
+
+class Listable1 l where
+  liftTiers :: [[a]] -> [[l a]]
```

Comment:
`Listable1` and `Listable2` lift `Listable` (the typeclass provided by LeanCheck; it’s like `Arbitrary`, only, well, not arbitrary) to kinds `* -> *` and `* -> * -> *` respectively. This is greatly convenient for defining `Listable` instances for fixpoint types like `Freer` and `Cofreer` without having to resort to `UndecidableInstances`.